modules/kubernetes/ingress-healthz/main.tf

```diff
@@ -31,6 +31,88 @@ resource "kubernetes_namespace" "this" {
   }
 }
 
+resource "kubernetes_network_policy" "deny_default" {
+  metadata {
+    name      = "deny-default"
+    namespace = kubernetes_namespace.this.metadata[0].name
+  }
+
+  spec {
+    pod_selector {
+      match_labels = {}
+    }
+
+    policy_types = ["Ingress", "Egress"]
+
+    ingress {
+      from {
+        pod_selector {}
+      }
+    }
+
+    egress {
+      to {
+        namespace_selector {}
+        pod_selector {
+          match_labels = {
+            k8s-app = "kube-dns"
+          }
+        }
+      }
+
+      ports {
+        port     = 53
+        protocol = "UDP"
```

Comment:
@phillebaba we should allow both UDP and TCP: https://datatracker.ietf.org/doc/html/rfc6891#section-4.3

Reply:
Not sure if this is something currently used by kube-dns so I need to look further into it.

Reply:
@phillebaba it should be used by all modern DNS servers, or else larger lookups (local or external) wouldn't work. I've witnessed it myself not working as expected with some versions of alpine.

If you want to check the code, kube-dns is using skydns: https://github.com/skynetservices/skydns/blob/master/server/server.go